pkg/kubeapiserver/authorizer/reload.go

```diff
@@ -173,6 +175,14 @@ func (r *reloadableAuthorizerResolver) newForConfig(authzConfig *authzconfig.Aut
 		case authzconfig.AuthorizerType(modes.ModeSystemMasters):
 			// no browsersafeauthorizer here becase that rewrites the resources.  This authorizer matches no matter which resource matches.
 			authorizers = append(authorizers, authorizerfactory.NewPrivilegedGroups(user.SystemPrivilegedGroup))
+		case authzconfig.AuthorizerType(modes.ModeMinimumKubeletVersion):
+			// Add MinimumKubeletVerison authorizer, to block a node from being able to access most resources if it's not new enough.
+			// We must do so here instead of in pkg/apiserver because it relies on a node informer, which is not present in generic control planes.
```

Comment:
This code runs for generic control planes (non-kube-apiserver) too, doesn't it? We are relying on no one configuring this mode in the authz config, and the only programs built from this forked package (as far as I know) are OCP's kube-apiserver, whose authz config should be provided by its operator, and Microshift, which is not managed by an operator.

Reply:
If the other authorizers are configured in [the authorizer config](https://github.com/openshift/kubernetes/commit/72a6eb500782a31b3f7246596d00ddba4a0b6a8d#diff-6c756a18b342480afacdf8ea8b4e150656d3117b06136f449c08b8047ef0d4aeR128) and can be enabled or disabled by an operator, is there a reason why the new authorizer shouldn't be in the same place and also configurable by an operator?

Reply:
No, I'm not arguing that at all. I'm pointing out that the comment says:

> We must do so here instead of in pkg/apiserver because it relies on a node informer, which is not present in generic control planes

But "doing so here" is not what makes it compatible with generic control planes, it's the fact that it is configurable. The comment is incomplete or misleading.